File: app/src/main/java/com/pedrosoares/cotacaoapp/data/preferences/ManagerPreferences.java
package com.pedrosoares.cotacaoapp.data.preferences;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;

public class ManagerPreferences {

    private static String LAYOUT_MANAGER = "LAYOUT_MANAGER";
    private static final String USER = "USER";

    private static SharedPreferences getSharedPreferences(@NonNull Context context) {
        return context.getSharedPreferences(USER, Context.MODE_PRIVATE);
    }

    public static String getLayoutManagerRecycler(Context context,String layoutManager){
        return getSharedPreferences(context).getString(LAYOUT_MANAGER, layoutManager);
    }

    public static void setLinearLayoutManager(Context context, String layoutManager){
        SharedPreferences.Editor editor = getSharedPreferences(context).edit();
        editor.putString(LAYOUT_MANAGER, layoutManager);
        editor.apply();
    }

    public static void setGridLayoutManager(Context context,String layoutManager){
        SharedPreferences.Editor editor = getSharedPreferences(context).edit();
        editor.putString(LAYOUT_MANAGER, layoutManager);
        editor.apply();
    }



}

File: app/src/main/java/com/pedrosoares/cotacaoapp/data/preferences/ManagerPreferences.kt
package com.pedrosoares.cotacaoapp.data.preferences

import android.content.Context
import android.content.SharedPreferences

object ManagerPreferences {
    private const val LAYOUT_MANAGER = "LAYOUT_MANAGER"
    private const val USER = "USER"
    private fun getSharedPreferences(context: Context): SharedPreferences {
        return context.getSharedPreferences(USER, Context.MODE_PRIVATE)
    }

    @JvmStatic
    fun getLayoutManagerRecycler(context: Context, layoutManager: String?): String {
        return getSharedPreferences(context).getString(LAYOUT_MANAGER, layoutManager)
    }

    @JvmStatic
    fun setLinearLayoutManager(context: Context, layoutManager: String?) {
        val editor = getSharedPreferences(context).edit()
        editor.putString(LAYOUT_MANAGER, layoutManager)
        editor.apply()
    }

    @JvmStatic
    fun setGridLayoutManager(context: Context, layoutManager: String?) {
        val editor = getSharedPreferences(context).edit()
        editor.putString(LAYOUT_MANAGER, layoutManager)
        editor.apply()
    }
}